concepts/modularization/README.md

```diff
@@ -0,0 +1,83 @@
+
+# Dependencies between components
+Components can depend only on core Kubernetes API or API extensions introduced by other components. Component operators are responsible for checking if the required APIs are available and react properly to the missing dependencies by reducing functionality or even reporting errors. Component owners are responsible for integration tests with their dependencies (with all versions supported in official release channels).
+
+**Example:**
+
+If the API you require is not available you should fail (e.g. core kubernetes API or istio virtual service). If your component can work without the API, but some features are not available (e.g. service monitor from monitoring) you should just skip it and continue to deploy other component resources. 
+
+# Release channels
+Release channels let customers balance between available features and stability. The number of channels and their characteristics can be established later. Usually, projects introduce between 2 and 4 channels. Some examples:
+- [GKE](https://cloud.google.com/kubernetes-engine/docs/concepts/release-channels)
+- [Nextcloud](https://nextcloud.com/release-channels/)
+
+TO DO:
+- how release channels are implemented (one config file or folder in some repository / config maps)
+- what component versions are included in the channel 
+- governance model - requirements for promoting version to the stable channel (hotfix should be possible anyway)
+
+# Versioning
+Kyma ecosystem produces several artifacts that can be deployed in the central control plane (KEB + operators) and in the target Kubernetes cluster. Versioning strategy should address pushing changes for all these artifacts in an unambiguous way with full traceability. Identified artifacts for each component
+- Operator CRD ([api/config/crd](https://github.com/kyma-project/manifest-operator/tree/main/api/config/crd))
+- Operator deployment ([operator/config/manager](https://github.com/kyma-project/manifest-operator/blob/main/operator/config/manager/manager.yaml))
+- Operator image (docker image in gcr)
+- Component CRDs ([installation/resources/crds](https://github.com/kyma-project/kyma/tree/main/installation/resources/crds))
+- Component deployment ([resources](https://github.com/kyma-project/kyma/tree/main/resources))
+- Component images (docker images in gcr)
+
+Component operators should be deployed continuously. Operators should support all versions that are currently available in all release channels. It is up to the component owner to decide how they manage different component versions inside operator (operator per version or single operator supporting multiple versions).
+
+## Simple versioning
+
+Simple versioning of component resources could be achieved by packaging component CRDs and charts into component operator binary (or container image). This way released operator would contain CRDs and charts of its components in the local filesystem. 
```

Comment:
Considering the need for compatibility I think it would not be easily possible this way to manage multiple versions for upgrade/downgrades between release channels or updates in an existing channel. (since the operator image would only know one chart at a time, If we start packaging multiple charts into the repository i imagine it will get quite messy and also somehow defeat the purpose of the chart already including a version, we can see this already in the current reconciler with istioctl being included in multiple versions in the image).

I think in the end we could think of completely decoupling the module version from the operator delivery. If developers change the chart they could use the normal compatibility in semver and on major upgrades also break compatibility for an operator in a given version. 

Then we could track the operator versions and component versions in two different release channel groups: one for "classic" SKR delivery (containing the kyma runtime modules, e.g. charts or resources) and one for "control-plane" delivery (containing the operators and CRD definitions) Using this we could end up with 2 release trains, with the control-plane delivery being decoupled from the SKR delivery: https://raw.githubusercontent.com/kyma-project/community/e7ff25edb3519ab1cc28ec0dee91578c67768c3d/concepts/operator-reconciliation/assets/channel_based_versioning.png

A component-operator could then "choose" how many versions and which versions to support for a given component without having to be imaged or versioned separately. This also makes the possibility of a security can easier since we will not have to rebuild images for fixes in the charts third party images.

Reply:
+1 that charts conceptually belong close to the delivery of an operator, still I think we should avoid having to package multiple versions of the same chart / data with different settings (see the Istioctl inclusion in the image on the current reconciler, this gets messy because you also have to triage security findings for these individually). That you would then have to package in the image. 

Regarding the verification of the charts: I suggest to anyhow introduce a BOM that is able to track digests of images and charts independently. Maybe we should consider something like https://github.com/gardener/landscaper/blob/master/docs/concepts/InstallationBlueprintRelationship.md#componentdescriptor-reference
This would also solve the security aspect as you are able to verify the integrity of channels or components through their respective part in the BOM (by comparing the digests) We will anyhow need this to ensure channel integrity / package integrity